astropy/utils/shapes.py

```diff
@@ -395,3 +397,70 @@ def unbroadcast(array):
     )
 
     return array.reshape(array.shape[first_not_unity:])
+
+
+def simplify_basic_index(basic_index, *, shape):
+    """
+    Given a Numpy basic index, return a tuple of integers and slice objects
+    with no default values (`None`) if possible.
+
+    If one of the dimensions has a slice and the step is negative and the stop
+    value of the slice was originally `None`, the new stop value of the slice
+    may still be set to `None`.
+
+    For more information on valid basic indices, see
+    https://numpy.org/doc/stable/user/basics.indexing.html#basic-indexing
+
+    Parameters
+    ----------
+    basic_index
+        A valid Numpy basic index
+    shape
+        The shape of the array being indexed
+    """
+    ndim = len(shape)
+
+    if not isinstance(basic_index, (tuple, list)):  # We just have a single int
+        basic_index = (basic_index,)
+
+    new_index = list(basic_index)
+
+    if Ellipsis in new_index:
+        if new_index.count(Ellipsis) > 1:
+            raise IndexError("an index can only have a single ellipsis ('...')")
+
+        # Replace the Ellipsis with the correct number of slice(None)s
+        e_ind = new_index.index(Ellipsis)
+        new_index.remove(Ellipsis)
+        n_e = ndim - len(new_index)
+        for i in range(n_e):
+            ind = e_ind + i
+            new_index.insert(ind, slice(None))
```

Comment:
I'd tend to also include a fully defined slice here, i.e., `slice(0, shape[ind], 1)` (to check: `ind` is correct index), since you do that below for appending slices.

Reply:
This does get tested if I configure hypothesis to run more examples, though I'm not sure if I should do that by default to avoid slowing down the test suite. I've also fixed your suggestion.

Reply:
I guess one could put some `hypothesis` examples in... but really overkill.